### PR TITLE
Fix ObjectUnderFileSystem isRoot to check stripped path

### PR DESCRIPTION
**Pull request by LuQQiu**
https://github.com/Alluxio/alluxio/issues/10264

In most of the ObjectUnderFileSystem operations, we will strip prefix if exist. Prefix here refers to the specific object UFS root key (s3://bucket_name for example). 

However, in our ObjectUnderFileSystem.isRoot check, we still check if the stripped path ("/" for example) equals to the full root key (s3://bucket_name).

This fix consider the isRoot case for both stripped path and original full url.

### CODE REVIEW

File: core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java [hunk 1]
```diff
@@ -886,8 +886,9 @@ protected String getParentPath(String path) {
    * @return true if the path is the root, false otherwise
```
Comment (ZacBlanco) [inline core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java]: perhaps add some javadoc with examples to this method to more clearly illustrate it's inputs/outputs?

**Comment by yuzhu** [comment]
```
Tests in error: 
alluxio.client.cli.fs.JobServiceFaultToleranceShellTest.distributedMv(alluxio.client.cli.fs.JobServiceFaultToleranceShellTest)
  Run 1: JobServiceFaultToleranceShellTest.before:52 » IO java.lang.RuntimeException: a...
  Run 2: JobServiceFaultToleranceShellTest.before:52 » IO java.lang.RuntimeException: a...

alluxio.client.fs.FreeAndDeleteIntegrationTest.deleteDir(alluxio.client.fs.FreeAndDeleteIntegrationTest)
  Run 1: FreeAndDeleteIntegrationTest.deleteDir:145 » TestTimedOut test timed out after...
  Run 2: FreeAndDeleteIntegrationTest.deleteDir:146 » TestTimedOut test timed out after...```

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by LuQQiu** [comment]
Tests in error: 
alluxio.server.ft.ZookeeperFailureIntegrationTest.zkFailure(alluxio.server.ft.ZookeeperFailureIntegrationTest)
  Run 1: ZookeeperFailureIntegrationTest.zkFailure:78 » Timeout Timed out waiting for a...
  Run 2: ZookeeperFailureIntegrationTest.zkFailure:78 » Timeout Timed out waiting for a...

alluxio.server.ft.journal.MultiMasterJournalTest.testTtl(alluxio.server.ft.journal.MultiMasterJournalTest)
  Run 1: MultiMasterJournalTest.before:53 » IO java.lang.RuntimeException: alluxio.exce...
  Run 2: MultiMasterJournalTest.testTtl:81 » IO java.lang.RuntimeException: alluxio.exc...

**Comment by LuQQiu** [comment]
jenkins, test this please

**Comment by LuQQiu** [comment]
Tests in error: 
alluxio.client.cli.fs.JobServiceFaultToleranceShellTest.distributedMv(alluxio.client.cli.fs.JobServiceFaultToleranceShellTest)
  Run 1: JobServiceFaultToleranceShellTest.before:52 » IO java.lang.RuntimeException: a...
  Run 2: JobServiceFaultToleranceShellTest.before:52 » IO java.lang.RuntimeException: a...

alluxio.client.fs.FileInStreamIntegrationTest.asyncCacheAfterSeek(alluxio.client.fs.FileInStreamIntegrationTest)
  Run 1: FileInStreamIntegrationTest.asyncCacheAfterSeek:508 » TestTimedOut test timed ...
  Run 2: FileInStreamIntegrationTest.asyncCacheAfterSeek:508 » TestTimedOut test timed ...

alluxio.client.fs.FreeAndDeleteIntegrationTest.deleteDir(alluxio.client.fs.FreeAndDeleteIntegrationTest)
  Run 1: FreeAndDeleteIntegrationTest.deleteDir:145 » TestTimedOut test timed out after...
  Run 2: FreeAndDeleteIntegrationTest.deleteDir:146 » TestTimedOut test timed out after...

alluxio.server.ft.journal.MultiMasterJournalTest.testTtl(alluxio.server.ft.journal.MultiMasterJournalTest)
  Run 1: MultiMasterJournalTest.before:53 » IO java.lang.RuntimeException: alluxio.exce...
  Run 2: MultiMasterJournalTest.before:53 » IO java.lang.RuntimeException: alluxio.exce...

**Comment by LuQQiu** [comment]
jenkins, test this please

**Comment by ZacBlanco** [comment]
alluxio-bot, merge this please

**Comment by ZacBlanco** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Comment by LuQQiu** [comment]
Fix #10264

**Comment by LuQQiu** [comment]
close #10264

**Pull request merged**